linux/dir_tools.bsh

```diff
@@ -470,4 +471,65 @@ function relative_path()
   else
     echo "${ans[*]}"
   fi
-}
\ No newline at end of file
+}
+
+#**
+# .. function:: strip_copy
+#
+# :Arguments: * ``$1`` - The source directory
+#             * ``$2`` - The destination directory
+#             * ``$3`` - Strip this many leading components from file names on ``cp``
+#
+# Replicates the behavior of ``tar --strip-components``, but when applied to a ``cp`` scenario.
+#
+# .. note:
+#
+#    ``cp`` will process files in a different order than ``tar --strip-components``. In case there are multiple files with the same name at a stripped depth,files may be overwritten differently than ``tar --strip-components`` would have. While ``tar`` keeps the first one in the archive order, ``cp`` will keep the last one in ``cp`` order. And as both order and behaviors are different, the result could be the same, or different.
+#**
+function strip_copy()
+{
+  local source_dir=$(real_path "${1}")
+  local destination=${2}
+  local depth=${3}
+
+  mkdir -p "${destination}"
+
+  # Technically the length of destination is not including in the xargs like
+  # formation internal the find command, however it turns out everyone's
+  # implementation of xargs/find adds such a large safety buffer that that
+  # discrepancy doesn't matter
+  #
+  # Newer versions of cp refused to overwrite a file it copied in the same
+  # command, "cp: will not overwrite just-created". However the behavior of
+  # --strip-components doesn't care, so we won't either
+  # This is slightly different from the tar behavior, where the first one is
+  # kept, while cp will override. But seeing as order isn't guaranteed to be
+  # maintained, this different in behavior hardly matters. Trying to disable
+  # this "error message" with --backup=none did not work, so just ignore cp
+  # exit code with || :
+  DEST="${destination}" find "${source_dir}" \
+                            -mindepth "${depth}" -maxdepth "${depth}" \
+                            -exec sh -c 'cp -fa "${@}" "${DEST}/" || :' sh \{\} \+
```

Comment:
Do you need to send stderr to /dev/null?

Reply:
I thought I would not. If I do that, and you have a real `cp` error like permissions or out of disk space, etc. Then it would be lost and very hard to debug (of course, we also lose the return value from this too now). So I  decided leave it unfiltered...